src/Frontend/Core.hs

```diff
@@ -29,6 +29,8 @@ module Frontend.Core
     , tabSelected
     , html
     , redirect
+    -- * Test only
+    , FormPage(..)
```

Comment:
This would be a good reason to move `Frontend.Core` to `Frontend.Core.Internal` and re-export everything from here *except* this.  Could you add this as a FIXME?